### ISSUE TITLE
OpenAPI gen throws errors on lazy schemas

### ISSUE THREAD

**Issue by 987Nabil** [zio/zio-http#2652]
No explicit schema definition in a companion object of a type `B`  that is used by a type `A`, that derives its schema, leads to the schema of `B` being lazy. This leads to an exception during OpenAPI generation.

```scala
case class A(b:B)

object A:
  given Schema[A] = DeriveSchema.gen

case class B(i: Int)
```

**Comment by sroopsai** [comment]
/attempt #2652

<div id="algora" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clrt18rif0004la0ffspkxoqs/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by 35C4n0r** [comment]
@jdegoes Is this issue still open, if not I would like to attempt it.

**Comment by jdegoes** [comment]
cc @987Nabil

**Comment by hochgi** [comment]
/attempt #2652

<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clrt18rif0004la0ffspkxoqs/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

Issue has been duplicated by #2933 and has already been fixed in #2935 
The opened PR ( #3004 ) simply adds a test with the example from above to show it has already been fixed.
